cmd/sanitize-prow-jobs/main.go

```diff
@@ -127,18 +149,34 @@ func main() {
 		os.Exit(0)
 	}
 
-	if len(opt.prowJobConfigDir) == 0 {
-		logrus.Fatal("mandatory argument --prow-jobs-dir wasn't set")
+	if err := opt.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options.")
 	}
-	if len(opt.configPath) == 0 {
-		logrus.Fatal("mandatory argument --config-path wasn't set")
+
+	dtEnforcer, err := opt.deprecateTemplates.GetEnforcer()
+	if err != nil {
+		logrus.WithError(err).Fatalf("Failed to load config from %q", opt.configPath)
 	}
 
 	config, err := dispatcher.LoadConfig(opt.configPath)
 	if err != nil {
 		logrus.WithError(err).Fatalf("Failed to load config from %q", opt.configPath)
 	}
-	if err := determinizeJobs(opt.prowJobConfigDir, config); err != nil {
+
+	if err := determinizeJobs(opt.prowJobConfigDir, config, dtEnforcer); err != nil {
 		logrus.WithError(err).Fatal("Failed to determinize")
 	}
+
+	if dtEnforcer.Enabled {
+		if err := dtEnforcer.SaveAllowlist(opt.deprecateTemplates.AllowlistPath); err != nil {
```

Comment:
Shouldn't this be behind another flag? I.E the allowlist will be only manually updated whereas the check if all template jobs are in the allowlist will eventually run in a presubmit